provider/pkg/gen/nodejs-templates/yaml/yaml.tmpl

```diff
@@ -383,9 +385,26 @@ export interface ConfigOpts {
     resourcePrefix?: string;
 }
 
-/** @ignore */ function yamlLoadAll(text: string, opts?: pulumi.ComponentResourceOptions): Promise<any[]> {
-    let invokeOpts: pulumi.InvokeOptions = { async: true, version: getVersion(), provider: opts?.provider };
+/** @ignore */ export function getChildOpts(parent: pulumi.Resource, opts?: pulumi.ComponentResourceOptions): pulumi.CustomResourceOptions {
+    return {
+        parent: parent,
+        ...opts?.version && {version: opts.version},
```

Comment:
There's a lot going on in this line of code. Is there a more verbose / less magical way to write this?

Reply:
I will add a comment to this effect: 
The goal is to conditionally declare a `version` property. The merge logic within `ResourceOptions` skips declared properties, even if their value is `undefined`.